### ISSUE TITLE
Warning about updated markdown syntax won't go away

### ISSUE THREAD

**Issue by mhkeller** [Gruntfuggly/todo-tree#471]
This alert shows up whenever I save a document, even after selecting "Never Show This Again."
<img width="461" alt="Screen Shot 2021-03-16 at 4 27 17 PM" src="https://user-images.githubusercontent.com/498744/111375366-89bb1680-8674-11eb-93c8-7ee201739046.png">

**Comment by Gruntfuggly** [comment]
Sorry - that must be annoying.  When you click "Never Show This Again" it should store a flag, so maybe this is failing for some reason. 

Please could you have a look in the developer console (Help -> Toggle Developer Tools) and see if there are any errors?

If you click Yes, does it still keep popping up?

**Comment by Gruntfuggly** [comment]
It should also only happen if the regex contains the legacy way to match markdown todos. You can fix this manually by changing the settings as described here: https://github.com/Gruntfuggly/todo-tree#markdown-support

**Comment by mhkeller** [comment]
I ended up updating to the new regex (at first I didn't bc it didn't seem to behave properly but now it works well)

**Comment by baincd** [comment]
@Gruntfuggly I think I may have discovered the issue.

This block of code
https://github.com/Gruntfuggly/todo-tree/blob/82f4ec8e253f70577300a54c759b20a3b0d94994/src/extension.js#L992-L995
needs `ignoreMarkdownUpdate = true;` added.

I noticed the following behavior:

I launched vscode with a new Global State (`code --user-data-dir ./todo-tree-test-user ./todo-tree-ws`), and open a markdown file, I keep seeing the popup even after clicking the "Never show this again" button.  

The reason is, the ignoreMarkdownUpdate was updated in the global state, but not the local variable.  After reloading the window, the popup is not displayed again.


![todo-tree-popup-issue](https://user-images.githubusercontent.com/5075659/123361487-2414db00-d53d-11eb-8a43-b89c461d2be0.png)
